server/src/test/java/org/elasticsearch/reservedstate/service/ReservedClusterStateServiceTests.java

```diff
@@ -450,15 +428,44 @@ public void testCheckMetadataVersion() {
         assertFalse(
             checkMetadataVersion("operator", operatorMetadata, new ReservedStateVersion(124L, Version.fromId(Version.CURRENT.id + 1)))
         );
-    }
 
-    private ReservedClusterStateHandler<Map<String, Object>> makeHandlerHelper(final String name, final List<String> deps) {
-        return new ReservedClusterStateHandler<>() {
-            @Override
-            public String name() {
-                return name;
-            }
+        ClusterState state = ClusterState.builder(new ClusterName("test")).metadata(Metadata.builder().put(operatorMetadata)).build();
```

Comment:
These are the additional tests added

Reply:
It's not clear to me what we are testing here; could you add some comments, or maybe separate this test into a different ones with better names, or add assertion messages so it's clear what we expect and why?

Reply:
Looking at it, the new checks subsume the old ones - so I've just removed the old checks.

This is basically testing that when the state version or release version is out of date, the cluster state is left as-is

Reply:
> testing that when the state version or release version is out of date, the cluster state is left as-is

I think that would be great for our future selves to add this as the assertion message.